docs/hyperopt.md

```diff
@@ -1,153 +1,109 @@
+            Categorical([True, False], name='rsi-enabled'),
+            Categorical(['bb_lower', 'macd_cross_signal'], name='trigger')
+        ]
 ```
-This says, "*one of the guards is RSI, it can have two values, enabled or 
-disabled. If it is enabled, try different values for it between 20 and 40*".
 
-So, the part of the strategy builder using the above setting looks like 
-this:
+Above definition says: I have five parameters I want you to randomly combine 
+to find the best combination. Two of them are integer values (`adx-value` 
+and `rsi-value`) and I want you test in the range of values 20 to 40.
+Then we have three category variables. First two are either `True` or `False`.
+We use these to either enable or disable the ADX and RSI guards. The last
+one we call `trigger` and use it to decide which buy trigger we want to use.
+
+So let's write the buy strategy using these values:
 
 ```
-if params['rsi']['enabled']:
-    conditions.append(dataframe['rsi'] < params['rsi']['value'])
+        def populate_buy_trend(dataframe: DataFrame) -> DataFrame:
+            conditions = []
+            # GUARDS AND TRENDS
+            if 'adx-enabled' in params and params['adx-enabled']:
+                conditions.append(dataframe['adx'] > params['adx-value'])
+            if 'rsi-enabled' in params and params['rsi-enabled']:
+                conditions.append(dataframe['rsi'] < params['rsi-value'])
+
+            # TRIGGERS
+            if params['trigger'] == 'bb_lower':
+                conditions.append(dataframe['close'] < dataframe['bb_lowerband'])
+            if params['trigger'] == 'macd_cross_signal':
+                conditions.append(qtpylib.crossed_above(
+                    dataframe['macd'], dataframe['macdsignal']
+                ))
+
+            dataframe.loc[
+                reduce(lambda x, y: x & y, conditions),
+                'buy'] = 1
+
+            return dataframe
+
+        return populate_buy_trend
 ```
 
-It checks if Hyperopt wants the RSI guard to be enabled for this 
-round `params['rsi']['enabled']` and if it is, then it will add a 
-condition that says RSI must be smaller than the value hyperopt picked 
-for this evaluation, which is given in the `params['rsi']['value']`.
-
-That's it. Now you can add new parts of strategies to Hyperopt and it 
-will try all the combinations with all different values in the search 
-for best working algo.
+Hyperopting will now call this `populate_buy_trend` as many times you ask it (`epochs`)
+with different value combinations. It will then use the given historical data and make
+buys based on the buy signals generated with the above function and based on the results
+it will end with telling you which paramter combination produced the best profits.
 
-
-### Add a new Indicators
+### Adding New Indicators
 If you want to test an indicator that isn't used by the bot currently, 
 you need to add it to the `populate_indicators()` method in `hyperopt.py`.
```

Comment:
i am not sure this segment makes sense - the current version only has very few indicators (which is fine for performance) ... so unless you hyperopt the default strategy you will always need to do this .

Reply:
Removed the title and added few more explaining words.